pilot/pkg/xds/sds.go

```diff
@@ -60,13 +60,18 @@ func (sr SecretResource) Cacheable() bool {
 	return true
 }
 
-func sdsNeedsPush(updates model.XdsUpdates) bool {
+func sdsNeedsPush(updates model.XdsUpdates, configStore model.ConfigStore) bool {
 	if len(updates) == 0 {
 		return true
 	}
-	if len(model.ConfigNamesOfKind(updates, gvk.Secret)) > 0 {
+	if model.ConfigsHaveKind(updates, gvk.Secret) {
 		return true
 	}
+	for conf := range model.ConfigsOfKind(updates, gvk.ReferencePolicy) {
+		if rp := configStore.Get(gvk.ReferencePolicy, conf.Name, conf.Namespace); isSecretReferencePolicy(rp) {
```

Comment:
`configStore.Get` is not safe here; everything in the push should only be computed from PushContext which is a point in time snapshot. Using the configstore directly leads to some race conditions that make config generation non-determinstic and break our cache

Reply:
Basically we can just check the type, which is what we do for all other `xxxNeedsPush`

Reply:
That's what I did in https://github.com/istio/istio/commit/a1e6330ec02134345c21554119dfb30248d5edba, would let @hzxuzhonghu confirm

Reply:
Yep, this is a superset,  not a big deal since this is not used widely